File: Creational-Patterns/Python/Factory/factory/image_factory.py
#!/usr/bin/env python
# coding=utf-8


from image import jpg,gif
from reader import gif_reader, jpg_reader

class ImageFactory(object):

    @staticmethod
    def read_image(pixels: list, img_type: str ):
        if( img_type == "gif" ):
            reader = gif_reader.GifReader();
        elif( img_type == "jpg" ):
            reader = jpg_reader.JPGReader();
        else:
            return None
        return reader.get_decoded_image( pixels );

File: Creational-Patterns/Java/Factory/src/factory/ImageFactory.java
package factory;

import reader.ImageReader;
import reader.GifReader;
import reader.JPGReader;

import image.AbstractImage;

public class ImageFactory {
    
    private  ImageReader reader;
   
    
    public ImageFactory() {
    }
    
    
    public AbstractImage readImage( int[] pixels, String type ){    
        if( type == "gif"){ 
            reader = new GifReader();
        } else if ( type == "jpg"){
            reader = new JPGReader();
        }else{
            return null;
        }
        return reader.getDecodedImage( pixels );
    }
}
